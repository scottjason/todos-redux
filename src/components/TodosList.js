import { connect } from 'react-redux';
import Todo from './Todo';
import AddTodo from './AddTodo';
import { editTodo } from '../redux/actions';
import './TodoList.scss';

const TodosList = props => {
  console.log(props)
  return (
    <main>
    <AddTodo />
      {props.todos.map(todo => {
        return <Todo key={todo._id} editTodo={props.editTodo} todo={todo} />;
      })}
    </main>
  );
};

const mapStateToProps = state => {
  const { todos } = state;
  return { todos };
};

const mapDispatchToProps = dispatch => {
  return {
    editTodo: todo => dispatch(editTodo(todo)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(TodosList);
