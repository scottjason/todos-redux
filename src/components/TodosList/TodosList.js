import React from 'react';
import { connect } from 'react-redux';
import Todo from '../Todo/Todo';
import AddTodo from '../AddTodo/AddTodo';
import { editTodo } from '../../redux/actions';
import './TodoList.scss';

const TodosList = props => {
  return (
    <main>
      <AddTodo />
      {props.todos.map(todo => {
        return <Todo key={todo._id} editTodo={props.editTodo} todo={todo} />;
      })}
    </main>
  );
};

const mapStateToProps = state => {
  const { todos } = state;
  return { todos };
};

const mapDispatchToProps = dispatch => {
  return {
    editTodo: todo => dispatch(editTodo(todo)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(TodosList);
