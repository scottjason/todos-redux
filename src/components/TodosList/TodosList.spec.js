import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import TodosList from './TodosList';

const middleware = [];
const initialState = {
  todos: [],
};
const mockStore = configureStore(middleware);
const store = mockStore(initialState);

describe('TodosList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <TodosList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
