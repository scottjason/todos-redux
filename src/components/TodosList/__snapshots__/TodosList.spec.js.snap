// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosList renders correctly 1`] = `
<main>
  <form
    onSubmit={[Function]}
  >
    <label
      className="visually-hidden"
      htmlFor="add-to-do"
    >
      add to do
    </label>
    <input
      autoComplete="off"
      className="add-todo"
      id="add-to-do"
      name="addTodo"
      onChange={[Function]}
      readOnly={false}
      type="text"
      value=""
    />
  </form>
</main>
`;
